core/src/main/groovy/noe/tomcat/configure/ConnectorConfiguratorTomcat.groovy

```diff
@@ -124,29 +124,33 @@ class ConnectorConfiguratorTomcat {
   }
 
   private void updateExistingConnector(Node connector, Node newConnector) {
-    // update attributes
+    // update connector attributes
     newConnector.attributes().each { attribute ->
       connector.@"${attribute.key}" = attribute.value
     }
 
-    // update nodes
     newConnector.each { Node newSubelement ->
-      String UpgradeProtocol = ConnectorUpgradeProtocolTomcat.ELEMENT_NAME
-
-      if (newSubelement.name() == UpgradeProtocol) {
-        if (connector.find { it.name() == UpgradeProtocol } == null) {
-          // create new element
-          connector.appendNode(newSubelement, newSubelement.attributes(), newSubelement.value())
-        } else {
-          // upgrade existing element
-          connector.findAll { it.name() == UpgradeProtocol }.each { upgradeProtocol ->
-            newSubelement.attributes() { attribute ->
-              upgradeProtocol.@"${attribute.key}" = attribute.value
-            }
-          }
-        }
+      updateInnerElements(connector, newSubelement)
+    }
+  }
+
+  /**
+   *
+   * @param connector
+   * @param newSubElement
+   *
+   * Search for existing inner element to remove and replace with new one
+   * Note: SSLHostConfig inner elements can exist more than one
+   */
+  private void updateInnerElements(Node connector, Node newSubElement) {
+    if (connector.find { it.name() == newSubElement.name() } != null) {
+      Node oldElement = (Node) connector.find { it.name() == newSubElement.name()}
+      if (oldElement.name() != "SSLHostConfig") {
```

Comment:
Not sure that I follow why SSLHostConfig has special care?

Reply:
There can be multiple SSLHostConfig elements inside one Connector.So if i would try to add second one it would be deleted and replaced from the new one

Reply:
> There can be multiple SSLHostConfig elements inside one Connector.

That is something that tomcat supports? How do we address that? How can one configure multiple SSLHostConfig elements inside one Connector form code?

Reply:
Isn't it discrepancy with https://github.com/web-servers/noe-core/pull/110/files/f4363f09c53f7ec1b3e01b149c423c44b749f7c5#r539279021?

It seems that something we want to set exactly as it is in code and something else means just an addition. I would consder unifying this approach.

Reply:
Although is possible that multiple SSLHostConfig elements can be present, do you want me to cut it down to only one? 
Tomcat has not document about this but i ve noticed many using it [1].

[1] https://wolfpaulus.com/tomcat-multiple-ssl-certificates/

Reply:
I wouldn't cut it down, what I am thinking is that the code shouldn't just add a new element because we don't know whether the client doesn't want to just update existing.

Since the attribute hostname seems as key, I would recommend considering
 * check if the element with key exists, 
     YES -> update it, 
     NO -> add new

WDYT?

Reply:
TBH first time I see hostname as attribute.I just shared the link as evidence of multiple SSLHostConfig usability.So I have nothing like key in order to follow your logic.